File: 파이썬/이코테/그리디/볼링공 고르기/Main2.py

N,M = map(int,input().split())
balls = list(map(int,input().split()))

# 서로 다른 무게의 볼링공을 선택해야함 공의 최대 무게 10
nums = [0]*(11)
for ball in balls:
    nums[ball]+=1

ans = 0

for i in range(1,11):
    if nums[i]!=0:
        ans+=nums[i]*sum(nums[i+1:])

print(ans)

'''
5 3
1 3 2 3 2

8 5
1 5 4 3 2 4 5 2
'''

File: Programmers/src/바킹독/이분탐색/용액/Main2.java
package 바킹독.이분탐색.용액;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;


public class Main2 {
    int N;
    int[] arr;

    public static void main(String[] args) throws IOException {
        new Main2().sol();
    }

    private void sol() throws IOException {
        setUp();
        int result = Integer.MAX_VALUE;
        int left = 0;
        int right = N - 1;
        String ans = "";
        while (left < right) {
            int sum = arr[left] + arr[right];
            if (Math.abs(sum) < result) {
                ans = String.format("%d %d", arr[left], arr[right]);
                result = Math.abs(sum);
            }
            if (sum >= 0) {
                right--;
            } else {
                left++;
            }
        }
        System.out.println(ans);
    }


    private void setUp() throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        N = Integer.parseInt(br.readLine());
        arr = new int[N];
        String[] input = br.readLine().split(" ");
        for (int i = 0; i < N; i++) {
            arr[i] = Integer.parseInt(input[i]);
        }
    }
}
